File: dungeons-26.1.2/src/main/java/fishmod/utils/JsonUtility.java
package fishmod.utils;

import fishmod.Bladeaddons;
import fishmod.utils.debug.Debug;
import fishmod.utils.dungeon.Split;
import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import org.jetbrains.annotations.NotNull;

import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;
import java.io.Reader;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.Map;

public class JsonUtility {

    public static @NotNull HashMap<String, ArrayList<Split>> readSplits(String path) {

        try (InputStream stream = Bladeaddons.class.getResourceAsStream(path)) {

            if (stream == null) return new HashMap<>();

            try (Reader reader = new InputStreamReader(stream)) {

                JsonElement element = JsonParser.parseReader(reader);
                return parseSplits(element);
            }
        } catch (IOException e) {
            Debug.LOGGER.error("Failed to parse a split");
        }

        return new HashMap<>();
    }

    private static HashMap<String, ArrayList<Split>> parseSplits(JsonElement jsonElement) {
        HashMap<String, ArrayList<Split>> floors = new HashMap<>();
        JsonObject object = jsonElement.getAsJsonObject();

        for (Map.Entry<String, JsonElement> floorEntry : object.entrySet()) {

            ArrayList<Split> splits = new ArrayList<>();

            String floorName = floorEntry.getKey();
            JsonArray dialoguesArray = floorEntry.getValue().getAsJsonArray();

            for (JsonElement dialogueElement : dialoguesArray) {
                JsonObject dialogueObj = dialogueElement.getAsJsonObject();

                int color = dialogueObj.get("color").getAsInt();
                String name = dialogueObj.get("name").getAsString();
                String start = dialogueObj.get("start").getAsString();
                String end = dialogueObj.get("end").getAsString();
                double avg = dialogueObj.has("avg") ? dialogueObj.get("avg").getAsDouble() : 0.0;

                splits.add(new Split(name, start, end, color, avg));
            }

            floors.put(floorName, splits);
        }
        return floors;
    }
}

File: dungeons-26.1.2/src/main/kotlin/fishmod/utils/JsonUtility.kt
package fishmod.utils

import com.google.gson.JsonElement
import com.google.gson.JsonParser
import fishmod.Bladeaddons
import fishmod.utils.debug.Debug
import fishmod.utils.dungeon.Split
import java.io.IOException
import java.io.InputStreamReader

object JsonUtility {

    @JvmStatic
    fun readSplits(path: String): HashMap<String, ArrayList<Split>> {
        try {
            Bladeaddons::class.java.getResourceAsStream(path).use { stream ->
                if (stream == null) return HashMap()
                InputStreamReader(stream).use { reader ->
                    val element = JsonParser.parseReader(reader)
                    return parseSplits(element)
                }
            }
        } catch (e: IOException) {
            Debug.LOGGER.error("Failed to parse a split")
        }

        return HashMap()
    }

    private fun parseSplits(jsonElement: JsonElement): HashMap<String, ArrayList<Split>> {
        val floors = HashMap<String, ArrayList<Split>>()
        val obj = jsonElement.asJsonObject

        for (floorEntry in obj.entrySet()) {
            val splits = ArrayList<Split>()

            val floorName = floorEntry.key
            val dialoguesArray = floorEntry.value.asJsonArray

            for (dialogueElement in dialoguesArray) {
                val dialogueObj = dialogueElement.asJsonObject

                val color = dialogueObj.get("color").asInt
                val name = dialogueObj.get("name").asString
                val start = dialogueObj.get("start").asString
                val end = dialogueObj.get("end").asString
                val avg = if (dialogueObj.has("avg")) dialogueObj.get("avg").asDouble else 0.0

                splits.add(Split(name, start, end, color, avg))
            }

            floors[floorName] = splits
        }
        return floors
    }
}
